File: p1/p1.py
import csv

FILENAME = "p1_input.txt"

def get_data():
    output = []
    with open(FILENAME, 'r') as fd:
        reader = csv.reader(fd)
        for row in reader:
            output.append(row)
    return output


def process(arr):
    totals = []
    subtotal = 0
    for r in arr:
        if not r:
            totals.append(subtotal)
            subtotal = 0
        else:
            subtotal += int(r[0])
    max = sum(sorted(totals, reverse=True)[:3])
    return max


def main():
    arr = get_data()
    max = process(arr)
    print(max)

main()

File: p1/p1.js
const fs = require('fs');

const FILENAME = "p1_input.txt"

const getData = (fn) => 
    fs.readFileSync(fn)
        .toString()
        .split("\n")

const findMax = (arr) => {
    let max = 0;
    let subTotal = 0;
    arr.forEach(el => {
        if (el === "") {  
            if (subTotal > max) {
                max = subTotal
            }
            subTotal = 0
        } else {
            subTotal += parseInt(el)
        } 
    });
    return max
    }

const findTopThree = (arr) => {
    let totals = [];
    let subTotal = 0;
    arr.forEach(el => {
        if (el === "") {  
            totals.push(subTotal)
            subTotal = 0
        } else {
            subTotal += parseInt(el)
        } 
    });
    const sArr = 
        totals
            .sort((a,b) => b-a)
            .slice(0,3)
            .reduce((a,b) => a+b, 0)
    return sArr
}




function main() {
    const data = getData(FILENAME);
    const max = findMax(data)
    const topThree = findTopThree(data) 
    console.log(topThree)
}

main()

